### PR TITLE
ci: upgrade to latest action version

### PR DESCRIPTION
**Pull request by wenytang-ms**

### CODE REVIEW

**Comment by wenytang-ms** [comment]
https://github.com/wenytang-ms/apicenter-sample/actions/runs/9850274632/job/27195498051

File: templates/.azure-pipelines/cd.yml [hunk 1]
```diff
@@ -15,4 +15,4 @@ steps:
       scriptType: "ps"
       scriptLocation: "inlineScript"
       inlineScript: |
-        az apic api register -g ${env:RESOURCE_GROUP} -s ${env:RESOURCE_NAME} --api-location ${env:FILE_LOCATION}
+        az apic api register -g ${env:RESOURCE_GROUP} -n ${env:RESOURCE_NAME} --api-location ${env:FILE_LOCATION}
```
Comment (formulahendry) [inline templates/.azure-pipelines/cd.yml:18]: install v1.0.0 of apic cli

File: templates/.github/cd.yml [hunk 2]
```diff
@@ -35,4 +35,4 @@ jobs:
         with:
           azcliversion: latest
           inlineScript: |
-            az apic api register -g ${{ vars.RESOURCE_GROUP }} -s ${{ vars.RESOURCE_NAME }} --api-location ${{ vars.FILE_LOCATION }}
+            az apic api register -g ${{ vars.RESOURCE_GROUP }} -n ${{ vars.RESOURCE_NAME }} --api-location ${{ vars.FILE_LOCATION }}
```
Comment (formulahendry) [inline templates/.github/cd.yml:38]: Same here. install v1.0.0 of apic cli

**Comment by wenytang-ms** [comment]
https://github.com/wenytang-ms/apicenter-sample/actions/runs/9851250895/job/27197796530

**Review by formulahendry — APPROVED** [review]

**Pull request merged**